src/NzbDrone.Core/Music/AlbumRepository.cs

```diff
@@ -301,20 +302,45 @@ public Album FindByTitle(int artistId, string title)
         }
 
         public Album FindByTitleInexact(int artistId, string title)
```

Comment:
Should this and `FindByStringInexact` be moved out to AlbumService.cs similar to how you did tracks? This really isnt calling to the db other than the query to get all artist albums which can be simplified like this from the service: `var albums = Query.Where(s => s.ArtistId == artistId).ToList();` can be changed to `var albums = _albumRepository.GetAlbums(artistId)`

Reply:
Might be sensible. I was also worrying whether the artist one might be slow for large libraries because it pulls all the artists from the database each time.  I could change it to just query a list of the clean names and operate on that, then having got the best match go back to the database for the full `Artist` class. Do you think that would be worth doing?

Reply:
I suppose we would have to benchmark it to see if it's making that big of a deal.